lib/shopify_dashboard_plus.rb

```diff
@@ -102,35 +130,43 @@ def validate_date_range(from, to)
 
 
   ## Metrics Helpers
-  def get_total_revenue(orders)
-    revenue = orders.collect{|order| order.total_price.to_f }.inject(:+).round(2) rescue 0
-    revenue ||= 0
+  def display_as_currency(value)
+    ShopifyAPI::Shop.current.money_with_currency_format.gsub("{{amount}}", value.to_s) rescue "N/A"
+  end
+
+  def get_date_range(first, last)
+    DateTime.parse(last).mjd - DateTime.parse(first).mjd
+  end
+
+  def get_total_revenue(collections)
+    # collections is an array of ActiveRecord::Collections, each containing orders
+    collections.collect{|orders| orders.collect{|order| order.total_price.to_f }.inject(:+).round(2) }.inject(:+).round(2) rescue 0
   end
   
-  def get_daily_revenues(start_date, end_date, orders)
+  def get_daily_revenues(start_date, end_date, collections)
     # Create hash entry for total interval over which to inspect sales
     revenue_per_day = {}
-    days = DateTime.parse(end_date).mjd - DateTime.parse(start_date).mjd
+    days = get_date_range(start_date, end_date)
     (0..days).each{ |day| revenue_per_day[(DateTime.parse(end_date) - day).strftime("%Y-%m-%d")] = 0 }
 
-    # Retreive orders between start and end date (up to 50)
-    revenue = orders.collect{|order| [order.created_at, order.total_price.to_f]}
+    # Retreive array of ActiveRecord::Collections, each containing orders between the start and end date
+    revenue = collections.collect{|orders| orders.collect{|order| [order.created_at, order.total_price.to_f]}}
     
     # Filter order details into daily totals and return
-    revenue.each do |sale|
-      revenue_per_day[DateTime.parse(sale[0]).strftime('%Y-%m-%d')] += sale[1]
+    revenue.each do |order|
+      order.each do |sale|
+        day_index = DateTime.parse(sale[0]).strftime('%Y-%m-%d')
+        revenue_per_day[day_index] = revenue_per_day[day_index].plus(sale[1])
+      end
     end
     revenue_per_day
   end
 
-  def hash_to_graph_format(sales)
+  def hash_to_graph_format(sales, merge_results = false)
```

Comment:
Keyword arguments might make it easier to understand what the second argument does when calling the method

```ruby
def hash_to_graph_format(sales, merge_results: false)
  # you now have access to a merge_results variable which defaults to false
end
```

which would look like

```
hash_to_graph_format(sales)
```

or 

```ruby
hash_to_graph_format(sales, merge_results: true)
```

when called.